src/components/sidecar/use-active-section.ts

```diff
@@ -1,18 +1,53 @@
 import { useEffect, useRef, useState } from 'react'
 import { SidecarHeading } from './types'
+import { useCurrentProduct } from 'contexts'
+import useCurrentPath from 'hooks/use-current-path'
+import getCSSVariableFromDocument from 'lib/get-css-variable-from-document'
 
 /**
- * This code comes from one of Bryce's (@BRKalow) @hashicorp/react-components PRs:
+ * The sticky header has a specific height and we care about headings that are
+ * visible below it. This function calculates the height of the header based on
+ * two CSS variables.
+ *
+ * TODO: this may need to be refactored when we address the brittleness of our
+ * header height.
+ */
+const getFullNavHeaderHeight = () => {
+  const navigationHeaderHeight = getCSSVariableFromDocument(
+    '--navigation-header-height',
+    { asNumber: true }
+  ) as number
+  const alertBarHeight = getCSSVariableFromDocument('--alert-bar-height', {
+    asNumber: true,
+  }) as number
+
+  return navigationHeaderHeight + alertBarHeight
+}
+
+/**
+ * This code started based off of one of Bryce's (@BRKalow) PRs in the
+ * @hashicorp/react-components repo:
  * https://github.com/hashicorp/react-components/pull/325
  */
 export function useActiveSection(
   headings: SidecarHeading[],
   isEnabled = true
 ): string {
+  const visibleHeadings = useRef<Set<string>>(new Set())
   const [activeSection, setActiveSection] = useState<SidecarHeading['slug']>()
   const previousY = useRef<number>()
 
+  // isProductLanding is needed to determine the IntersectionObserver threshold
+  // because the headings on product landing pages are smaller than the on docs
+  // pages
+  const currentPath = useCurrentPath({ excludeHash: true, excludeSearch: true })
+  const currentProduct = useCurrentProduct()
+  const isProductLanding = currentPath === `/${currentProduct.slug}`
```

Comment:
I don't _love_ this, and maybe this calls for revisiting why headings have different sizes between the product landing pages and docs-y content pages, but I was getting really tired of testing out a bunch of values to only have them work on either a product landing page or it's docs-y content.